src/main/kotlin/uk/gov/justice/digital/hmpps/visitallocationapi/config/WebClientConfiguration.kt

```diff
@@ -1,17 +1,83 @@
 package uk.gov.justice.digital.hmpps.visitallocationapi.config
 
+import io.netty.resolver.DefaultAddressResolverGroup
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.context.annotation.Bean
 import org.springframework.context.annotation.Configuration
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
+import org.springframework.http.codec.ClientCodecConfigurer
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.ExchangeStrategies
 import org.springframework.web.reactive.function.client.WebClient
+import reactor.netty.http.client.HttpClient
+import reactor.netty.resources.ConnectionProvider
 import uk.gov.justice.hmpps.kotlin.auth.healthWebClient
 import java.time.Duration
+import java.time.Duration.ofSeconds
 
 @Configuration
 class WebClientConfiguration(
   @Value("\${hmpps-auth.url}") val hmppsAuthBaseUri: String,
   @Value("\${api.health-timeout:2s}") val healthTimeout: Duration,
+  @Value("\${prisoner.search.url}") private val prisonSearchBaseUrl: String,
+  @Value("\${incentives.api.url}") private val incentivesBaseUrl: String,
 ) {
+  private enum class HmppsAuthClientRegistrationId(val clientRegistrationId: String) {
+    PRISONER_SEARCH("other-hmpps-apis"),
+    INCENTIVES("other-hmpps-apis"),
+  }
+
+  @Bean
+  fun hmppsAuthHealthWebClient(builder: WebClient.Builder): WebClient {
+    return builder.healthWebClient(hmppsAuthBaseUri, healthTimeout)
+  }
+
+  @Bean
+  fun prisonerSearchWebClient(authorizedClientManager: OAuth2AuthorizedClientManager): WebClient {
+    val oauth2Client = getOauth2Client(authorizedClientManager, HmppsAuthClientRegistrationId.PRISONER_SEARCH.clientRegistrationId)
```

Comment:
We could do this in a more concise way now as per the kotlin template - https://github.com/ministryofjustice/hmpps-template-kotlin/blob/main/src/main/kotlin/uk/gov/justice/digital/hmpps/templatepackagename/config/WebClientConfiguration.kt